nixos/modules/services/misc/navidrome.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.navidrome;
+  configFile = "/etc/navidrome/settings.json";
+  dataFolder = "/var/lib/navidrome";
+in {
+  options = {
+
+    services.navidrome = {
+      enable = mkEnableOption "Enable Navidrome music server and streamer";
+
+      settings = lib.mkOption {
+        type = types.attrs;
+        default = {};
+        example = {
+          LogLevel = "INFO";
+          BaseURL = "/music";
+          ScanInterval = "10s";
+          TranscodingCacheSize = "15MB";
+          MusicFolder = "/Media/Music";
+        };
+        description = ''
+        Configuration for Navidrome, see <link xlink:href="https://www.navidrome.org/docs/usage/configuration-options/"/> for supported values.
+      '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "navidrome";
+        description = "User account under which Navidrome runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "navidrome";
+        description = "Group account under which Navidrome runs.";
+      };
+
+      musicFolderPermissions = mkOption {
+        type = types.str;
+        default = "775";
+        description = ''
+        The permissions to set for the music folder.
+        They will be applied on every service start.
+        '';
+      };
+
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.settings.MusicFolder}' '${cfg.musicFolderPermissions}' '${cfg.user}' '${cfg.group}' - -"
+    ];
+
+    environment.etc."navidrome/settings.json".text = builtins.toJSON (cfg.settings // {
+      # These settings are always overwritten by NixOS i.e. they are not settable
+      DataFolder = dataFolder;
+      ConfigFile = configFile;
+    } // (if (attrsets.hasAttrByPath ["Port"] cfg.settings)
```

Comment:
This is exactly what the `optionalAttrs` function does, you should use it.

Reply:
I didn't know, now I use it. Thank you!